### ISSUE TITLE
New Huobi withdraw problem

### ISSUE THREAD

**Issue by KindSpidey** [ccxt/ccxt#16016]
Hello guys. As I see, you finally merged Huobi Withdraw PR (https://github.com/ccxt/ccxt/pull/15885), but it creates another problem. I try to withdraw now 
`huobi.withdraw("BBF", amount, address, tag=None, params={"network": "ERC20"})`
and get error
`AttributeError: 'huobipro' object has no attribute 'networkPairIdFromCode'`
How to fix it?

**Comment by ttodua** [comment]
Thanks @KindSpidey for report, the fix will be applied asap, there was missed typo mechanical mistake in last merge.

**Comment by KindSpidey** [comment]
@ttodua got it and wait for it

**Comment by KindSpidey** [comment]
@ttodua @karlosmiei
Hey guys. Is this fix in ccxt now? Because I still get this error, trying to withdraw
 `AttributeError: 'huobipro' object has no attribute 'networkPairIdFromCode'`
Version of ccxt - 2.2.91

**Comment by ttodua** [comment]
you will get further updates for network after https://github.com/ccxt/ccxt/pull/16024 is merged.

**Comment by KindSpidey** [comment]
> > @ttodua @carlosmiei Hey guys. Is this fix in ccxt now? Because I still get this error, trying to withdraw `AttributeError: 'huobipro' object has no attribute 'networkPairIdFromCode'` Version of ccxt - 2.2.91
> 
> the latest update (2.2.91) seems to include the updated code, see it: https://github.com/ccxt/ccxt/blob/master/js/huobi.js#L5037
> so, there is some issue on your end, seems you dont have the updated code. maybe you use some kind of caching mechanism, or ... please revise your userland codes.

Ok, got it. Thanks

**Comment by KindSpidey** [comment]
@ttodua deleted and install ccxt again - same error. Installed via PyCharm.
Error sends me to 4725 row in "huobi" file of ccxt
       ```
 if networkCode is not None:
            request['chain'] = self.networkPairIdFromCode(code, networkCode)
```
error is the same
`AttributeError: 'huobipro' object has no attribute 'networkPairIdFromCode'`
Version is `2.2.91`

**Comment by KindSpidey** [comment]
@ttodua well, are you sure python version of Huobi gets updated? Because js version has update 21 hours ago, but Python version only 3 days ago. Compare
JS - https://github.com/ccxt/ccxt/blob/master/js/huobi.js
Python - https://github.com/ccxt/ccxt/blob/master/python/ccxt/huobi.py

**Comment by carlosmiei** [comment]
@KindSpidey yeah probably the build failed, and the package was not published, will take a look asap.

**Comment by KindSpidey** [comment]
@ttodua @carlosmiei as I see now, huobi.py got commit, so I try to test and make withdraw 
`huobi.withdraw("BBF", amount, address, tag=None, params={"networkCode": "ERC20"})`
and get error
`ccxt.base.errors.BadRequest: huobipro {"status":"error","err-code":"base-currency-chain-error","err-msg":"The current currency chain does not exist","data":null}`

**Comment by ttodua** [comment]
@carlosmiei my bad, the last commit where you corrected `getNetworkPairIdFromCode` into `networkCodeToId`, I missed to tell that the last unified method needed `networkCode` as first argument, and `code` as second argument.
Anyway, in linked PR https://github.com/ccxt/ccxt/pull/16024, I've fixed that issue too.

**Comment by KindSpidey** [comment]
@ttodua ok, thanks, when should this Huobi/OKX PR arrived? And have you checked exact issue with coin BBF network ERC20 withdraw from Huobi?

**Comment by ttodua** [comment]
> @ttodua ok, thanks, when should this Huobi/OKX PR arrived? And have you checked exact issue with coin BBF network ERC20 withdraw from Huobi?

yes, I have checked BBF too, the final fixes will be submitted today and probably within 24 hours it might get merged, if @carlosmiei has time.

**Comment by KindSpidey** [comment]
@ttodua thank you very much. So what can you say about BBF? Example of my code doesn't work in your too?

**Comment by ttodua** [comment]
yes, in my top first PR there I had missed a typo, which causes the issue till date, and there are pending PR's what fix the issue.
about BBF, with new PR, you will be able to withdraw using your code well (your code was correct).

**Comment by KindSpidey** [comment]
@ttodua ok, I'm so happy to hear that, because I thought it was mine, Huobi or exact this coin problem. Thank you very much for fixing and feedbacks. Wait for releasing PR

**Comment by KindSpidey** [comment]
Hello @ttodua. While PRs aren't merged, can you provide example of code, how I can withdraw money from Huobi (and especially BBF) now? What should I use in network param? "bbftrc20" or "trc20bbf"? How?

**Comment by carlosmiei** [comment]
Hello @KindSpidey, the `networkCodeToId` issue will be fixed immediately, so it does not block you, other optimizations will be done separately

**Comment by KindSpidey** [comment]
> Hello @KindSpidey, the `networkCodeToId` issue will be fixed immediately, so it does not block you, other optimizations will be done separately 

ok, can you tag me, when its done?

**Comment by KindSpidey** [comment]
@carlosmiei oh, it is already done i see

**Comment by carlosmiei** [comment]
@KindSpidey yeah but the build was not released yet will probably take 30min +-

**Comment by KindSpidey** [comment]
@carlosmiei ok, I hope python version will also get merge

**Comment by KindSpidey** [comment]
@carlosmiei 

ready now?

**Comment by KindSpidey** [comment]
@ttodua and @carlosmiei withdraw works now, even BBF withdraw. Thank you very much guys!

**Comment by carlosmiei** [comment]
> 

@KindSpidey thank you for all the feedback provided! 🙂

**Comment by kroitor** [comment]
Closing this one during a cleanup — it was actually resolved and confirmed working back in Dec 2022 (see above 🎉) and simply never got closed. For anyone landing here from search: the `networkPairIdFromCode` method no longer exists — network handling was long since unified via `networkCodeToId`, and the exchange itself is now `htx` in ccxt following Huobi's rebrand. Thanks again @KindSpidey for the thorough back-and-forth back then! 🙏
